lib/facter/framework/core/fact_loaders/fact_loader.rb

```diff
@@ -20,8 +20,12 @@ def initialize
 
     def load(user_query, options)
       @internal_facts = load_internal_facts(user_query, options)
-      @custom_facts = load_custom_facts(options)
-      @external_facts = load_external_facts(options)
+      if ENV['INSIDE_FACTER']
+        @log.debug('INSIDE_FACTER env var detected, not loading custom/external facts off to prevent recursion')
+      else
```

Comment:
I'm thinking it would be better to set `ENV['INSIDE_FACTER'] = true` here (prior to loading custom/external facts) instead of in the `execute` method, in case a custom fact did something like:

```ruby
Facter.add(:recurse) do
  setcode { `facter os` }
done
```

The environment variable `INSIDE_FACTER` will automatically be passed to any/all child process no matter which ruby method is used to create the process (spawn, backticks, etc).